### PR TITLE
feat(strm-1259): update driver for keycloak auth

### PR DESCRIPTION
**Pull request by bobvandenhoogen**

### CODE REVIEW

File: strmprivacy/driver/client/auth.py [hunk 1]
```diff
@@ -10,60 +10,57 @@
...
+    _headers = {"content-type": "application/x-www-form-urlencoded; charset=UTF-8"}
 
-    def __init__(self, purpose: str, billing_id: str, client_id: str, client_secret: str, config: ClientConfig):
+    def __init__(self, purpose: str, client_id: str, client_secret: str, config: ClientConfig):
         """
         Service responsible for ensuring the validity of access tokens for the StrmPrivacyClient
 
         :param purpose: for which is the AuthService initialized
-        :param billing_id: unique customer identifier
-        :param client_id: unique stream identifier
```
Comment (ivan-p92) [inline strmprivacy/driver/client/auth.py:21]: client id is still a param, right?

File: strmprivacy/driver/client/auth.py [hunk 2]
```diff
@@ -10,60 +10,57 @@
...
-        return self.auth_provider.id_token
+        return self.auth_provider.access_token
 
-    def _authenticate(self, billing_id: str, client_id: str, client_secret: str) -> None:
+    def _authenticate(self, client_id: str, client_secret: str) -> None:
         self._logger.debug("authenticate")
         try:
-            payload = AuthRequest(billing_id, client_id, client_secret)
-            self._do_post(self._config.sts_auth_uri, payload)
+            payload = f"grant_type=client_credentials&client_id={client_id}&client_secret={client_secret}"
```
Comment (ivan-p92) [inline strmprivacy/driver/client/auth.py:41]: is this a payload, or query params?

File: strmprivacy/driver/domain/config.py [hunk 3]
```diff
@@ -15,11 +15,10 @@ def __init__(self, log_level=logging.INFO, **kwargs):
...
         self._egress_health_endpoint = kwargs.get("egress_health_endpoint", "/is-alive")
-        self._sts_protocol = kwargs.get("sts_protocol", "https")
-        self._sts_host = kwargs.get("sts_host", "sts.strmprivacy.io")
-        self._sts_auth_endpoint = kwargs.get("sts_auth_endpoint", "/auth")
-        self._sts_refresh_endpoint = kwargs.get("sts_refresh_endpoint", "/refresh")
-        self._sts_refresh_interval = kwargs.get("sts_refresh_interval", 3300)
+        self._keycloak_protocol = kwargs.get("keycloak_protocol", "https")
+        self._keycloak_host = kwargs.get("keycloak_host", "accounts.strmprivacy.io")
+        self._keycloak_endpoint = kwargs.get("keycloak_endpoint", "/auth/realms/streams/protocol/openid-connect/token")
+        self._keycloak_refresh_interval = kwargs.get("sts_refresh_interval", 300)
```
Comment (ivan-p92) [inline strmprivacy/driver/domain/config.py:21]: The property name should be updated

**Comment by ivan-p92** [comment]
An additional general comment: should we use `keycloak` at all in naming? We have abstracted it away, so we don't need to "expose" our implementation per se.

File: strmprivacy/driver/client/auth.py (hunk 2 above)
Comment (bobvandenhoogen) [reply strmprivacy/driver/client/auth.py:44]: It is the actual payload of the request

**Review by ivan-p92 — APPROVED** [review]

**Pull request merged**